accepted/future-releases/nnbd/feature-specification.md

```diff
@@ -516,6 +541,12 @@ implicit casts) shall succeed with a warning whenever the runtime type test
 would have succeeded if all `?` types were ignored, `Never` were treated as
 `Null`, and `required` named parameters were treated as optional.
 
+In legacy libraries, whether run in strong or weak checking mode, instance
+checks (`e is T`) also return `false` whenever they would have done so pre-nnbd.
+Specifically, if `e` evaluates to `null`, then `e is T` always returns `false`
+unless `T` is `Null`, `Never`, `Object`, `dynamic`, `void`, or a nullable or
+legacy version of any of the above.
```

Comment:
Would it suffice to say 'unless `T` is a type whose value is `Null`, `Never`, or a legacy top type'?

Otherwise it's not obvious that it includes cases like `e is X` where `X` is a type variable whose value is `Never`, or `e is FutureOr<Object>`, and a bunch of other cases.

Reply:
Is it that simple?

If we do `null is FutureOr<Never>` then that's equivalent to `null is FutureOr<Never*>` in a legacy library, but not to `null is FutureOr<Never*>*`.  Which is true, but it's not a top type.

So: `null is T` returns true if `T` is _Null_, _Never_, a top type, _S?_ for any _S_, or `FutureOr<..>` of any type where it returns true?

Maybe:

> When `e is T` occurs in a legacy library, if `e` evaluates to `null` and the legacy interpretation of the type expression `T` is _S*_, then the test evaluates to `true` if _Null_ <: _S_ or  _S_ is _Never_, and to `false` otherwise.

I wouldn't try to enumerate the infinitely many top types.
I'm actually a little surprised that `null is Never` succeeds. Are we sure we want that?

Reply:
How about `T <: Null` or `Object <: T`?

> I'm actually a little surprised that `null is Never` succeeds. Are we sure we want that?

We certainly need `null is Never*` for backwards compatibility, and I don't see much reason to explicitly exclude `Never`.